File: android/src/main/java/com/getcapacitor/community/classes/events/MessageIDEvent.java
package com.getcapacitor.community.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import java.util.UUID;

public class MessageIDEvent {

    @NonNull
    UUID messageID;

    public MessageIDEvent(@NonNull UUID messageID) {
        this.messageID = messageID;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        result.put("messageID", messageID.toString());

        return result;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Events/MessageIDEvent.swift
import Foundation
import Capacitor

@objc public class MessageIDEvent: NSObject {
    let messageID: UUID

    init(_ messageID: UUID) {
        self.messageID = messageID
    }

    public func toJSObject() -> JSObject {
        var result = JSObject()

        result["messageID"] = messageID.uuidString.lowercased()

        return result
    }
}
